src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -871,18 +871,26 @@ private void mapVendorAccount(List<CompletableFuture<?>> futures, Organization o
       Document doc) {
     Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_ACCOUNT))
       .ifPresent(vendorAccountField -> vendorAccountField.resolve(doc)
-        .thenAccept(vendorAccountNo -> Optional.ofNullable(organization.getAccounts().get(0).getAccountNo())
-          .ifPresentOrElse(organizationAccountNo -> {
-            if (vendorAccountNo.equals(HelperUtils.extractSubAccount(organizationAccountNo))) {
+        .thenAccept(vendorAccountNo -> Optional.ofNullable(organization.getAccounts().stream()
+         .map(Account::getAccountNo).collect(Collectors.toList()))
+         .ifPresentOrElse(organizationAccountNo -> {
+          if (!HelperUtils.getVendAccountFromOrgAccountsList((String) vendorAccountNo,organizationAccountNo).isEmpty()) {
```

Comment:
Please call the method **HelperUtils.getVendAccountFromOrgAccountsList((String) vendorAccountNo** only once (after 876 line) and then reuse this method in the 877 and 879 lines